docs/constrained.rst

```diff
@@ -0,0 +1,126 @@
+Constrained optimization
+========================
+
+This section is concerned with problems of the form
+
+.. math::
+
+    \min_{x} f(x, \theta) \textrm{ subject to } x \in \mathcal{C}(\upsilon),
+
+where :math:`f(x, \theta)` is a smooth function, :math:`x` are the parameters
```

Comment:
I hope people use it on things like relu that are not technically smooth. I would replace smooth with "functions for which we can compute its gradient"

Reply:
Same for other files